test/basic_test.cc

```diff
@@ -108,15 +108,30 @@ void BM_KeepRunning(benchmark::State& state) {
 BENCHMARK(BM_KeepRunning);
 
 void BM_KeepRunningBatch(benchmark::State& state) {
-  // Choose a prime batch size to avoid evenly dividing max_iterations.
-  const benchmark::IterationCount batch_size = 101;
+  // Choose a batch size >1000 to skip the typical runs with iteration
+  // targets of 10, 100 and 1000.  If these are not actually skipped the
+  // bug would be detectable as consecutive runs with the same iteration
+  // count.  Below we assert that this does not happen.
+  const benchmark::IterationCount batch_size = 1009;
```

Comment:
can you retain the comment that this needs to be prime to avoid the even division please?

Reply:
I removed the comment because I don't think it applies any more. If I understand the original intent correctly, the prime batch size was to help ensure that `max_iters` was not an even multiple of the batch size.  With the new approach, `max_iters` will immediately become a multiple of the batch size, whatever it is.  E.g. with a batch size of 17 `max_iters` will go from 1 to 17 to 170, etc.  Only the final 1.4x estimate might fail to be a multiple, but does a prime batch size makes that any more likely?

With this new way of growing `max_iters`, I could not think of a way to continue to exercise the not-evenly-divisible code paths explicitly, so I removed the comment instead.  It would be nice to retain that test coverage, so I'm open to ideas.  My only idea was to crack open some of these helper functions to make them more explicitly testable, but I didn't want to start off with that big of a change.